### ISSUE TITLE
Profiles not working in pip package

### ISSUE THREAD

**Issue by aarongorka** [ThreatResponse/aws_ir#71]
The current version available in pip3 package (`aws-ir-0.3.0`) doesn't work with either `--profile` or an assume-role equivalent. It fails on the initial bucket creation step:

```$ aws_ir --dry-run --verbose --profile myprofile --examiner-cidr-range '4.4.4.4/32' instance-compromise --target 8.8.8.8
Traceback (most recent call last):
  File "/home/aaron.gorka/Documents/venv-aws_ir/bin/aws_ir", line 10, in <module>
    c.run()
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/aws_ir/cli.py", line 182, in run
    self.config.profile
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/aws_ir/libs/case.py", line 53, in __init__
    self.case_bucket = self._setup_bucket(region='us-west-2')
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/aws_ir/libs/case.py", line 167, in _setup_bucket
    self.s3_resource
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/aws_ir/libs/s3bucket.py", line 12, in __init__
    self.bucket = self.find_or_create_by()
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/aws_ir/libs/s3bucket.py", line 20, in find_or_create_by
    bucket = self._create_s3_bucket()
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/aws_ir/libs/s3bucket.py", line 52, in _create_s3_bucket
    'LocationConstraint': self.region
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/botocore/client.py", line 317, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/aaron.gorka/Documents/venv-aws_ir/lib64/python3.6/site-packages/botocore/client.py", line 615, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (AccessDenied) when calling the CreateBucket operation: Access Denied
```

The same error is returned if I omit the `--profile` flag and use [awsume](https://github.com/trek10inc/awsume) to get access keys for a role.

After installing from the master branch, both methods work fine.

**Comment by andrewkrug** [comment]
Hello @aarongorka !  Thanks for reporting on this.  This is due to the fact that the role you're assuming doesn't have createBucket permissions.  We'd recommend creating a bucket and passing your case bucket to the command.  If the bucket already exists in the account the tool will not try to create one.

**Comment by aarongorka** [comment]
Hey @andrewkrug, thanks for the quick reply. My account does not have permissions to create a bucket, but the role I was trying to assume definitely had permissions to create a bucket, confirmed by the fact that installing the latest commit from master (I just cloned locally and ran `pip install ./aws_ir`) allowed me to run `instance-compromise` without any issues. I also manually tested permissions with `aws s3 mb ...`. I actually started to debug the issue and got pretty deep in to it before I realised it was already fixed in master (lesson learnt the hard way 😆). 

FWIW, removing the arguments from `boto3.setup_default_session(profile_name=self.profile)` in the pip version allowed me to assume role correctly.
